pkg/golden/go.mod

```
@@ -0,0 +1,30 @@
+module golden
```

```suggestion
module github.com/open-telemetry/opentelemetry-collector-contrib/pkggolden
```

Pretty sure this is causing a lot of the problems.